examples/reflection/generic_reflection.rs

```diff
@@ -12,8 +12,10 @@ fn main() {
         .run();
 }
 
+/// The `#[derive(Reflect)]` macro will automatically add any required bounds to `T`,
+/// such as `Reflect` and `GetTypeRegistration`.
```

Comment:
Is there a way to opt out of this behaviour (e.g. if you have a `PhantomData<T>`)? If so, it may be nice to also mention that here.

Reply:
this is done by default.

Reply:
Actually, this example is a bit misleading. The reflection bounds are only added to active fields now. Generic type parameters only get the `TypePath + Send + Sync + 'static` bounds I believe.

So the actual way to opt-out would be to `#[reflect(ignore)]` the field or add `#[reflect(no_field_bounds)]` (probably in combination with `#[reflect(where)]`). 

But I think that's outside the scope of this PR and should probably be tackled on its own since it might require renaming/changing up the example itself.

Reply:
I see, yeah it does get more complex than what should be in this example. Still, the wording of the docs make it seems like there's something not intuitive going on.

I wouldn't consider this a blocker for the PR, though it would be nice if this can be improved now or in the future.